src/vs/workbench/contrib/chat/browser/contrib/chatInputCompletions.ts

```diff
@@ -591,6 +595,36 @@ class BuiltinDynamicCompletions extends Disposable {
 		this._register(CommandsRegistry.registerCommand(BuiltinDynamicCompletions.addReferenceCommand, (_services, arg) => this.cmdAddReference(arg)));
 	}
 
+	private findActiveCodeEditor(): ICodeEditor | undefined {
+		const codeEditor = this.codeEditorService.getActiveCodeEditor();
+		if (codeEditor) {
+			const model = codeEditor.getModel();
+			if (model?.uri.scheme === Schemas.vscodeNotebookCell) {
+				return undefined;
+			}
+
+			if (model) {
+				return codeEditor;
+			}
+		}
+		for (const codeOrDiffEditor of this.editorService.getVisibleTextEditorControls(EditorsOrder.MOST_RECENTLY_ACTIVE)) {
+			let codeEditor: ICodeEditor;
+			if (isDiffEditor(codeOrDiffEditor)) {
```

Comment:
fyi there is a utility method called `getCodeEditor` in `src/vs/editor/browser/editorBrowser.ts`

Reply:
Sure. This is your code from https://github.com/microsoft/vscode/blob/c2b5fd631b414873257c141f55bd3ffb56c38ebc/src/vs/workbench/contrib/chat/browser/contrib/chatImplicitContext.ts#L135 :)